tests/plugins/filter_evaluation.test.ts

```diff
@@ -164,4 +166,35 @@ describe("Filter Evaluation Plugin", () => {
     });
     expect(model.getters.getFilter("sh2", 0, 0)).toBeTruthy();
   });
+
+  test("Inserting rows above or below the data filter header updates the filtered rows", () => {
+    const model = new Model();
+
+    createFilter(model, "A1:A3");
+    setCellContent(model, "A2", "Hi");
+    setCellContent(model, "A3", "Hi");
+
+    updateFilter(model, "A1", ["Hi"]);
+    addRows(model, "before", 0, 1);
```

Comment:
for both tests ,check after/before and use the getter `isRowFiltered`, to ensure that it is the work of the filter itself.